0.14.5: macOS ADIOS1 Linker Error

The latest backports introduced the following ADIOS1 linker error on macOS (x86_64):
```
    [ 10%] Linking CXX shared library ../lib.macosx-10.9-x86_64-3.6/openpmd_api/libopenPMD.ADIOS1.Serial.dylib
    Undefined symbols for architecture x86_64:
      "openPMD::internal::operationAsString(openPMD::Operation)", referenced from:
          openPMD::ADIOS1IOHandlerImpl::flush() in ADIOS1IOHandler.cpp.o
    ld: symbol(s) not found for architecture x86_64
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
    make[2]: *** [../lib.macosx-10.9-x86_64-3.6/openpmd_api/libopenPMD.ADIOS1.Serial.dylib] Error 1
```

First seen in #1286
Also seen in https://github.com/conda-forge/openpmd-api-feedstock/pull/88

I think we need to add `src/IO/Task.cpp` to `IO_ADIOS1_SEQUENTIAL_SOURCE` and `IO_ADIOS1_SOURCE` in `CMakeLists.txt`:
https://github.com/openPMD/openPMD-api/blob/0.14.5/src/IO/IOTask.cpp

We also need to check if we neeed to add `OPENPMDAPI_EXPORT` to the function in `include/openPMD/IO/IOTask.cpp`:
https://github.com/openPMD/openPMD-api/blob/0.14.5/include/openPMD/IO/IOTask.hpp#L68

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -488,13 +488,15 @@ set(IO_ADIOS1_SEQUENTIAL_SOURCE
         src/auxiliary/Filesystem.cpp
         src/ChunkInfo.cpp
         src/IO/ADIOS/CommonADIOS1IOHandler.cpp
-        src/IO/ADIOS/ADIOS1IOHandler.cpp)
+        src/IO/ADIOS/ADIOS1IOHandler.cpp
+        src/IO/IOTask.cpp)
 set(IO_ADIOS1_SOURCE
         src/Error.cpp
         src/auxiliary/Filesystem.cpp
         src/ChunkInfo.cpp
         src/IO/ADIOS/CommonADIOS1IOHandler.cpp
-        src/IO/ADIOS/ParallelADIOS1IOHandler.cpp)
+        src/IO/ADIOS/ParallelADIOS1IOHandler.cpp
+        src/IO/IOTask.cpp)
 
 # library
 if(openPMD_BUILD_SHARED_LIBS)

